ssl/t1_enc.c

```diff
@@ -375,6 +379,38 @@ int tls1_change_cipher_state(SSL *s, int which)
     if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
         goto skip_ktls;
 
+# ifdef __FreeBSD__
+    memset(&crypto_info, 0, sizeof(crypto_info));
+    if (EVP_CIPHER_mode(c) == EVP_CIPH_GCM_MODE) {
+	 crypto_info.cipher_algorithm = CRYPTO_AES_NIST_GCM_16;
+	 crypto_info.iv_len = EVP_GCM_TLS_FIXED_IV_LEN;
+    } else if (EVP_CIPHER_mode(c) == EVP_CIPH_CBC_MODE) {
+         if (s->ext.use_etm)
+             goto skip_ktls;
+	 crypto_info.cipher_algorithm = CRYPTO_AES_CBC;
+	 crypto_info.iv_len = EVP_CIPHER_iv_length(c);
+	 switch (EVP_CIPHER_nid(c)) {
+	 case NID_aes_128_cbc_hmac_sha1:
+	 case NID_aes_256_cbc_hmac_sha1:
+	      crypto_info.auth_algorithm = CRYPTO_SHA1_HMAC;
+	      break;
+	 case NID_aes_128_cbc_hmac_sha256:
+	 case NID_aes_256_cbc_hmac_sha256:
+	      crypto_info.auth_algorithm = CRYPTO_SHA2_256_HMAC;
```

Comment:
So what if we're not using one of the "stitched" ciphersuites (e.g. because the platform we're on doesn't provide assembler support for them)? We might still be using SHA1/SHA2-256, but we seem to go to skip_ktls in that case.